src/ucp/wireup/address.c

```diff
@@ -637,12 +613,10 @@ static void *ucp_address_unpack_md_info(const void *ptr,
                                         int *empty_dev)
 {
     uint8_t md_byte = *(uint8_t*)ptr;
-    size_t mask;
 
     *empty_dev = md_byte & UCP_ADDRESS_FLAG_MD_EMPTY_DEV;
-    mask       = ucp_address_md_mask(addr_version);
-
-    return ucp_address_unpack_byte_extended(ptr, mask, addr_version, md_index);
+    return ucp_address_unpack_byte_extended(ptr, UCP_ADDRESS_FLAG_MD_MASK,
+                                            addr_version, md_index);
```

Comment:
there seems to be still a wire-compat issue - when old address arrives, we will initialize md_index as invalid

Reply:
What if when we parse the address, we check if 2 MDs are equal in the low 5 bits, and if yes - assume the remote peer packed 64-bit memory domain indexes?
Also, when packing address, if have more than 32 MDs on the context, and we are packing only some of them (because of tl_bitmap) - packe another empty devices with a memory domain index that will cause a "collision" to indicate a new version

Reply:
There are 2 issues this this PR:
1) need to check why wire compatibility tests passed (it should fail)
2) need to check ucx address release version to calculate `md_mask` like
```
addr_md_mask = UCP_ADDRESS_FLAG_MD_MASK;
if ((ucp_address_pack_release_version() > ucp_address_unpack_release_version()) {
    addr_md_mask |= UCP_ADDRESS_FLAG_MD_ALLOC | UCP_ADDRESS_FLAG_MD_REG;
}
```
@yosefe will it work for the case you described?

Reply:
1. IMO the issue of wire compant manifests as rndv protocol falls back to active message: we unpack address with wrong md index, then get rkey with correct md_index, but since they differ we think we don't have valid uct rkeys. maybe there is no "fatal" failure in case we unpacked wrong md_index.
2. unfortuantely seems address v1 does not have version info in the header, only the unused address v2:
 https://github.com/openucx/ucx/blob/dda920bce756e9673836dc1e0903cc987159f729/src/ucp/wireup/address.c#L1135

Reply:
for v1 it is packed here
https://github.com/openucx/ucx/blob/dda920bce756e9673836dc1e0903cc987159f729/src/ucp/wireup/address.c#L1507

Reply:
cool, so even better, we can use it